executor/runtime/docker/docker_linux.go

```diff
@@ -150,6 +145,11 @@ func setupSystemServices(parentCtx context.Context, c runtimeTypes.Container, cf
 			continue
 		}
 
+		if svc.PullFailed {
```

Comment:
Do we really need a new field here?

Can we not use `svc.ContainerName` ? If it is unset (==""), then the volume container didn't get created for whatever reason, so is unsuitable to start.

Reply:
Right now the name is set in the middle of the function, and the actual `ContainerCreate` call could fail and we'd still wind up in the same situation. I'll move setting it to the end and drop the field.